diff --git a/src/analysis/CMakeLists.txt b/src/analysis/CMakeLists.txt
index 8a45e2229f..2c773ccd65 100644
--- a/src/analysis/CMakeLists.txt
+++ b/src/analysis/CMakeLists.txt
@@ -456,7 +456,7 @@ ENDIF(HAVE_OPENCL)
 #############################################################
 # qgis_analysis library
 
-ADD_LIBRARY(qgis_analysis SHARED ${QGIS_ANALYSIS_SRCS} ${QGIS_ANALYSIS_HDRS})
+ADD_LIBRARY(qgis_analysis ${LIBRARY_TYPE} ${QGIS_ANALYSIS_SRCS} ${QGIS_ANALYSIS_HDRS})
 
 GENERATE_EXPORT_HEADER(
    qgis_analysis
diff --git a/src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp b/src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp
index d9e54ac3e6..16187aeeb2 100644
--- a/src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp
+++ b/src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp
@@ -123,6 +123,7 @@ QgsLayoutAtlasToImageAlgorithm *QgsLayoutAtlasToImageAlgorithm::createInstance()
 
 QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
 {
+#ifndef QT_NO_PRINTER
   // this needs to be done in main thread, layouts are not thread safe
   QgsPrintLayout *l = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
   if ( !l )
@@ -248,6 +249,9 @@ QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap
   QVariantMap outputs;
   outputs.insert( QStringLiteral( "FOLDER" ), directory );
   return outputs;
+#else
+  return QVariantMap();
+#endif
 }
 
 ///@endcond
diff --git a/src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp b/src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp
index cd60f0ee82..025b7bfa28 100644
--- a/src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp
+++ b/src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp
@@ -125,6 +125,7 @@ QgsLayoutAtlasToPdfAlgorithm *QgsLayoutAtlasToPdfAlgorithm::createInstance() con
 
 QVariantMap QgsLayoutAtlasToPdfAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
 {
+#ifndef QT_NO_PRINTER
   // this needs to be done in main thread, layouts are not thread safe
   QgsPrintLayout *l = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
   if ( !l )
@@ -234,6 +235,9 @@ QVariantMap QgsLayoutAtlasToPdfAlgorithm::processAlgorithm( const QVariantMap &p
   QVariantMap outputs;
   outputs.insert( QStringLiteral( "OUTPUT" ), dest );
   return outputs;
+#else
+  return QVariantMap();
+#endif
 }
 
 ///@endcond
diff --git a/src/analysis/processing/qgsalgorithmlayouttoimage.cpp b/src/analysis/processing/qgsalgorithmlayouttoimage.cpp
index d99dd61412..ffd97e4018 100644
--- a/src/analysis/processing/qgsalgorithmlayouttoimage.cpp
+++ b/src/analysis/processing/qgsalgorithmlayouttoimage.cpp
@@ -115,6 +115,7 @@ QgsLayoutToImageAlgorithm *QgsLayoutToImageAlgorithm::createInstance() const
 
 QVariantMap QgsLayoutToImageAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
 {
+#ifndef QT_NO_PRINTER
   // this needs to be done in main thread, layouts are not thread safe
   QgsPrintLayout *l = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
   if ( !l )
@@ -184,6 +185,9 @@ QVariantMap QgsLayoutToImageAlgorithm::processAlgorithm( const QVariantMap &para
   QVariantMap outputs;
   outputs.insert( QStringLiteral( "OUTPUT" ), dest );
   return outputs;
+#else
+  return QVariantMap();
+#endif
 }
 
 ///@endcond
diff --git a/src/analysis/processing/qgsalgorithmlayouttopdf.cpp b/src/analysis/processing/qgsalgorithmlayouttopdf.cpp
index 4eba130a02..4a5a33a285 100644
--- a/src/analysis/processing/qgsalgorithmlayouttopdf.cpp
+++ b/src/analysis/processing/qgsalgorithmlayouttopdf.cpp
@@ -120,6 +120,7 @@ QgsLayoutToPdfAlgorithm *QgsLayoutToPdfAlgorithm::createInstance() const
 
 QVariantMap QgsLayoutToPdfAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
 {
+#ifndef QT_NO_PRINTER
   // this needs to be done in main thread, layouts are not thread safe
   QgsPrintLayout *l = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
   if ( !l )
@@ -196,6 +197,9 @@ QVariantMap QgsLayoutToPdfAlgorithm::processAlgorithm( const QVariantMap &parame
   QVariantMap outputs;
   outputs.insert( QStringLiteral( "OUTPUT" ), dest );
   return outputs;
+#else
+  return QVariantMap();
+#endif
 }
 
 ///@endcond
